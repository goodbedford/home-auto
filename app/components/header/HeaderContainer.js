import React from "react";
import {Route, Link, Redirect} from "react-router-dom";
import Header from "./Header";
import httpHelper from "../../utils/httpHelper.js";

const propTypes =  {
};
const defaultProps = {
};

class HeaderContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      date: new Date().toDateString()
    };
  }
  componentDidMount () {

  }

  render() {
    return (
      <Header
        date={this.state.date}
      />
    );
  }
}
HeaderContainer.propTypes = propTypes;
HeaderContainer.defaultProps = defaultProps;

export default HeaderContainer;
