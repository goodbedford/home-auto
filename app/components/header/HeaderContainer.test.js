/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import HeaderContainer from "./HeaderContainer";


describe("<HeaderContainer />", () => {

  it("HeaderContainer snapshot test", () => {
    const wrapper = shallow(<HeaderContainer />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should have a date prop", () => {
    const date = jest.fn( () => "01111979");
    const wrapper = shallow(<HeaderContainer />);
    wrapper.setState({date: "01111979"});
    expect(toJson(wrapper).props.date).toBe("01111979");
  });


});
