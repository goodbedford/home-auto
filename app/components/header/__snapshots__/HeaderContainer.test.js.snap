// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> HeaderContainer snapshot test 1`] = `
<Header
  date="Wed Apr 05 2017"
/>
`;
